crates/proto/src/rr/resource.rs

```diff
@@ -264,6 +264,51 @@ impl Record {
     pub fn into_data(self) -> RData {
         self.rdata
     }
+
+    /// Consumes `Record` and returns its components
+    pub fn into_parts(self) -> RecordParts {
+        self.into()
+    }
+}
+
+/// Consumes `Record` giving public access to fields of `Record` so they can
+/// be destructured and taken by value
+pub struct RecordParts {
+    /// label names
+    pub name_labels: Name,
+    /// record type
+    pub rr_type: RecordType,
+    /// dns class
+    pub dns_class: DNSClass,
+    /// time to live
+    pub ttl: u32,
+    /// rdata
+    pub rdata: RData,
+    /// mDNS cache flush
+    #[cfg(feature = "mdns")]
+    pub mdns_cache_flush: bool,
+}
+
+impl From<Record> for RecordParts {
+    fn from(record: Record) -> Self {
+        let Record {
+            name_labels,
+            rr_type,
+            dns_class,
+            ttl,
+            rdata,
+            mdns_cache_flush,
```

Comment:
looks like this also needs the mdns feature flag, see test failure.

Reply:
Maybe I just don't use cargo features very often, but this doesn't compile:
```
  let Query {
            name,
            query_type,
            query_class,
            #[cfg(feature = "mdns")]
            mdns_unicast_response,
        } = q;
        ```

Reply:
ugh, yeah, I was worried about that. this is annoying because mDNS is still not usable, but gets in the way.

You can use this construct:

```
cfg_if::cfg_if! {
            if #[cfg(feature = "mdns")] {
                ...
            } else {
                ...
            }
}
```

Reply:
You're current changes are also fine.